### PR TITLE
chore(opentelemetry-operator): upgrade opentelemetry-operator subchart

### PR DESCRIPTION
**Pull request by mat-rumian**
##### Description
(cherry picked from commit d4d75994a1c31f9daabca8b25281b81ca9a301f8)

This PR:
* updates OpenTelemetry Operator subchart to latest version
* moves `Instrumentation` CR generation to confs (config map created for this purpose)
* adds Job which applies Instrumentation CRs to specified namespaces
* disables by default creation of `Instrumentation` resource - `opentelemetry-operator.createDefaultInstrumentation=false`
* adds `opentelemetry-operator.instrumentationNamespaces` parameter for `Instrumentation` resource creation 

---

##### Checklist

- [x] Changelog updated

###### Testing performed

- Java instrumentation with version 1.16.0
- Python instrumentation with version 0.28b0
- NodeJS instrumentation with version 0.27.0
- .NET instrumentation with version 0.3.1-beta.1
- self-signed certificate creation
- Instrumentation CR creation (new Job)

### CODE REVIEW

File: deploy/helm/sumologic/README.md [hunk 1]
```diff
@@ -303,8 +303,8 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
-| `opentelemetry-operator.manager.env.WATCH_NAMESPACE`                      | Used to set value for `WATCH_NAMESPACE` environment variable which specifies Namespace to watch and create Instrumentation objects.                                                                                                                                                                                                                                  | `Nil`                                                                                                             |
+| `opentelemetry-operator.createDefaultInstrumentation`                     | Flag to control creation of default Instrumentation object                                                                                                                                                                                                                                                                                                           | `false`                                                                                                           |
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/README.md:306]: Isn't this kind of a breaking change? :sweat_smile:

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -5108,10 +5108,14 @@ tailing-sidecar-operator:
...
-  createDefaultInstrumentation: true
+
+  ## Specific for Tracing - Instrumentation resource creation
+  createDefaultInstrumentation: false
+  instrumentationNamespaces: ""
+
   admissionWebhooks:
     failurePolicy: Fail
-    enabled: false
+    enabled: true
```
Comment (astencel-sumo) [inline deploy/helm/sumologic/values.yaml:5118]: Isn't this a breaking change?

File: deploy/helm/sumologic/README.md (hunk 1 above)
Comment (perk-sumo) [reply deploy/helm/sumologic/README.md:306]: Yeah, that seems like a breaking change to me to.

File: deploy/helm/sumologic/values.yaml (hunk 2 above)
Comment (mat-rumian) [reply deploy/helm/sumologic/values.yaml:5118]: It's not as in version before admissionWebhooks were created by us with certificates.

File: deploy/helm/sumologic/README.md (hunk 1 above)
Comment (mat-rumian) [reply deploy/helm/sumologic/README.md:306]: Could be a breaking change, but we were treating OT Operator for Tracing as `BETA`. Appropriate comments in the next PR with changelog and release description will be added about this change.

**Review by astencel-sumo — APPROVED** [review]

**Pull request merged**